packages/@aws-cdk/aws-codepipeline-actions/lib/lambda/invoke-action.ts

```diff
@@ -86,8 +69,8 @@ export class LambdaInvokeAction extends codepipeline.Action {
       },
       configuration: {
         FunctionName: props.lambda.functionName,
-        UserParameters: props.userParameters
-      }
+        UserParameters: JSON.stringify(props.userParameters),
```

Comment:
I think you might need to use `this.node.stringifyJson` instead so tokens will be rendered properly. Also, add a test that verifies that tokens work.

Reply:
Done.